code/headers/gcode_generator.hpp

```diff
@@ -0,0 +1,163 @@
+#pragma once
+#include <hwlib.hpp> // size_t, (u)intx_t
+#include <vector3.hpp>
+
+namespace r2d2::robot_arm {
+    /**
+     * Template Interface class for generating gcodes.
+     * Has a few helper functions such as converting an int to a char.
+     * Size represents the size of the char buffer array.
+     * Note that a string of 4 characters, has a size of 5 with the '\0'
+     * character
+     * @param Size buffer size
+     * */
+    template <size_t Size>
+    class gcode_generator_c {
+    private:
+        /**
+         * Represents the size of the string currently in buffer
+         * */
+        size_t str_len = 0;
+
+        /** Checks whether the string to be added fits into the buffer
+         * @true if it fits
+         * @return false otherwise
+         * */
+        bool string_fits(const char *string) {
+            return get_string_length(string) + 1 + str_len <= Size;
+        }
+
+        /**
+         * Reverses a string
+         *
+         * @param char*
+         * @param int length
+         * */
+        void reverse(char *string, int length) {
+            int start = 0;
+            length--;
+            while (start < length) {
+                std::swap(*(string + start), *(string + length));
+                start++;
+                length--;
+            }
+        }
+
+    protected:
+        /**
+         * Buffer with Size as size.
+         * */
+        char buffer[Size] = {'\0'};
+
+        /**
+         * Converts an int to a string.
+         * Please note that the destination size must be 11 if you
+         * want to store a max integer value (2.147.483.647) + '\0' = 11
+         *
+         * @param int axis
+         * @param char array
+         * */
+        void int_to_string(int axis, char *string) {
```

Comment:
Is axis a good name for converting a int to a string? It's very limited to only be able to use a axis to convert to a string.